pkg/daemon/daemon.go

```diff
@@ -539,8 +540,25 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 	}
 
 	if reqReboot {
-		glog.Info("nodeStateSyncHandler(): reboot node")
-		rebootNode()
+		if err := wait.PollImmediateUntil(3*time.Second, func() (bool, error) {
+			if utils.ClusterType == utils.ClusterTypeOpenshift {
+				cmc, _ := dn.node.Annotations[mcdconst.CurrentMachineConfigAnnotationKey]
+				dmc, _ := dn.node.Annotations[mcdconst.CurrentMachineConfigAnnotationKey]
+				mcdState, _ := dn.node.Annotations[mcdconst.MachineConfigDaemonStateAnnotationKey]
+				if cmc == "" || cmc != dmc || mcdState != mcdconst.MachineConfigDaemonStateDone {
+					glog.Info("nodeStateSyncHandler(): node is not ready for reboot, wating for MCO state Done")
+					return false, nil
+				}
+			}
+
+			glog.Info("nodeStateSyncHandler(): reboot node")
+			rebootNode()
```

Comment:
Not sure if because the pr is still WIP, so feel free to ignore the comment. Should pause the pool here and unpause it afterwards?

Reply:
No, the PR doesn't want to pause the pool. The target is only preventing SNO from interrupting MCD when it's configuring the node.

Reply:
umm, I thought the solution we were looking for was for a regular cluster. Even for SNO case, if you don't pause the pool nothing is stopping from MCO rendering and applying a new MC if admin initiated one.

Reply:
also, note that for SNO case, MCO is going to skip drain operation, see https://github.com/openshift/machine-config-operator/pull/2457

Reply:
Right, like sinny said SNO should be a separate consideration

Reply:
It is ok for SNO to be interrupted by MCO. It can continue its work after MCO reboot the node. SNO doesn't support a single node cluster yet.